api/src/main/java/pl/kolendateam/dadcard/characterCard/entity/Character.java

```diff
@@ -1,29 +1,158 @@
+        this.classPgArray = new ArrayList<>();
+        this.savingThrow = new SavingThrow();
+        this.classSkills = new ArrayList<>();
+    }
+    
+    public void addClassToPgArray(ClassPg classPg) {
+        this.classPgArray.add(classPg);
+    }
+
+    public void incrementLevelClassForIndex(int index){
+        this.getClassPgArray().get(index).incrementLevel();
+    }
+
+    public void incrementLep(){
+        this.lep +=1;
+    }
+
+    public void addSTLevelOne(ClassPg classPg){
+
+        String stringSavingThrow = classPg.getSavingThrow();
+
+            double bonus;
+            if(stringSavingThrow.charAt(0) == 'h'){
+                bonus = 2.5;
+            } else{bonus = 0;}
+            this.getSavingThrow().addSTFortitude(bonus);
+
+            if(stringSavingThrow.charAt(1) == 'h'){
+                bonus = 2.5;
+            } else{bonus = 0;}
+            this.getSavingThrow().addSTReflex(bonus);
+
+            if(stringSavingThrow.charAt(2) == 'h'){
+                bonus = 2.5;
+            } else{bonus = 0;}
+            this.getSavingThrow().addSTWill(bonus);            
+        }
+
+    public void incementST() {
+        this.getSavingThrow().incementSTFortitude();
+        this.getSavingThrow().incementSTReflex();
+        this.getSavingThrow().incementSTWill();
+    }
+
+    public void setSkillsTruePgArray(Set<Skills> availableSkills) {
+        for(Skills skill : availableSkills){
+            for(ClassSkills classSkill : classSkills){
+                if(skill.getName().equals(classSkill.getNameSkill())){
+                    classSkill.setClassSkill(true);
+                }
+            }
+        }
+    }
+
+    public void createSkillsArray(List<Skills> skillsList) {
+
+        for(int x = 0; x < skillsList.size(); x++){
+
+            ClassSkills skill = new ClassSkills();
+            skill.setNameSkill(skillsList.get(x).getName());
+            skill.setClassSkill(false);
+            skill.setSkillRank(0);
+            
+            this.classSkills.add(skill);
+        }
+    }
+
+    public void calculateSkillPointsFirstLevel(int skPoints) {
+        this.skillPoints = skPoints * 3;
+    }
+
+    public void calculateSkillPoints(int skPoints) {
+        this.skillPoints += skPoints;
+    }
+
+    public void buySkills(String nameSkill, int lep, int skPoints) {
+        for(ClassSkills skill : classSkills){
+            if(skill.getNameSkill().equals(nameSkill)){
+                boolean check = true;
+                if(skPoints > lep+3){
+                check = false;
+                }
+                if(skill.isClassSkill()==true && skill.getSkillRank()>=lep+3){
+                check = false;
+                }
+                double doubleLEP = (lep+3)/2;
+                if(skill.isClassSkill()==false && skill.getSkillRank()>=(int)doubleLEP){
+                check = false;
+                }
+                if(check == true){
+                    if(skill.isClassSkill()==true){
```

Comment:
warunki cos == true sa bez sensu

Reply:
> warunki cos == true sa bez sensu

Nawet jeżeli zmienna jest booleen? Muszę wiedzieć czy jest to skill klasowy czy nie, wtedy nadam mu true